src/journal.rs

```diff
@@ -193,6 +190,41 @@ impl Journal {
         Ok(journal)
     }
 
+    /// Open the systemd journal located in a specific folder for reading.
+    ///
+    /// Params:
+    ///
+    /// * path: the absolute directory path. All journal files in this directory
+    ///   will be opened and interleaved automatically.
+    /// * files: the set of journal files to read. If the calling process
+    ///   doesn't have permission to read the system journal, a call to
+    ///   `Journal::open` with `System` or `All` will succeed, but system
+    ///   journal entries won't be included. This behavior is due to systemd.
+    /// * os_root: if true, journal files are searched for below the usual
+    ///   /var/log/journal and /run/log/journal relative to the specified path,
+    ///   instead of directly beneath it.
+    pub fn open_directory(path: &str, files: JournalFiles, os_root: bool) -> Result<Journal> {
```

Comment:
Can `path` here be `&Path`? Or does that cause some issue/reduce efficiency?

Reply:
The reformatting was caused by my environment (vscode). You could easily drop it and just copy the few lines that I added. That way it is cleaned up.

With regards to the exposed structs: I don't know if there is a rust specific guideline. In a quick search I just found this: https://deterministic.space/elegant-apis-in-rust.html#what-makes-an-api-elegant. At the end it is sematics: "let x = systemd::Journal" reads better than "let x = systemd::journal::Journal". On the other end: "systemd::journal::send()" reads perfectly fine. If your library would not be used by anyone yet, you could also easily avoid to publish the struct twice. Unfortunately that could be a breaking change, so it would be quite unfriendly to those that use your library. That's clearly something up to you. Maybe consider such change for the next major release if at all.

The path should actually be a &Path rather than a &str. You are right. I don't think there can be a huge performance issue.

Reply:
The naming `systemd::Journal` does seem nice, and is a very good reason to re-export them. 👍